### ISSUE TITLE
Drag and drop to reorder rows

### ISSUE THREAD

**Issue by kvenkata1974** [pavankataria/SwiftDataTables#40]
Hi,
I'm using version 0.8 of SwiftDataTables.

Can the rows not be reorder via drag and drop?

Was hoping the delegate would allow for drag and drop of rows.

Cheers
Karthik

**Comment by kvenkata1974** [comment]
I should be using the drag and drop delegate of the collection view, will try that first.

**Comment by kvenkata1974** [comment]
Sorry, tried using UICollectionViewDelegate and implemented the methods (extended SwiftDataTable since it already implements the protocol):

    func collectionView(_ collectionView: UICollectionView, canFocusItemAt indexPath: IndexPath) -> Bool {
        return true
    }
    
    func collectionView(_ collectionView: UICollectionView, moveItemAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
        print("Starting Index: \(sourceIndexPath.item)")
        print("Ending Index: \(destinationIndexPath.item)")
    }

But I could not get reorder of items via drag and drop to be triggered. Please help.

**Comment by pavankataria** [comment]
Yeah I didn't have this in mind and not sure how adopting those methods will affect the ordering of the swift data tables datasource. Come to think of it, I believe you'll need to update the datasource yourself once you figure out where the user wants to drag and drop an item. You'll then need to refresh the collection view to reflect the changes.

**Comment by pavankataria** [comment]
Thanks for this feature request! I've migrated all feature requests to the new [voting page](https://swiftdatatables.pavankataria.com/vote) where you can upvote this and other features.

This helps prioritise what I should work on next based on community interest. Please vote if you'd like to see this implemented!

Closing this issue as feature requests are now tracked on the website.
